### DISCUSSION TITLE
How to create models that work consistently

### DISCUSSION THREAD

**Discussion by hadronknights** [FIRST-Tech-Challenge/FtcRobotController#518] [Q&A] [answered]
We have been using ftc ml tool chain to develop models to detect team signals this season. We have tried multiple types of signals (objects) but we are unable to detect the signal consistently (maybe once in 10 attempts) during autonomous. Any recommendation how we can make the model more reliable. Is light an issue? If so, teams will never have lighting condition similar to tournament, therefore model created in a team practice area from school libraries to garages has to work in the tournament fields. Looking for expert guidance here on how to improve the object detection.

FYI: FTC provided model Powerplay.tflite works consistently though.

**Comment by texasdiaz** [comment, marked as answer]
Hey there! I'll take that as a compliment for the default model - that required quite a bit of patience, trial and error, and exploring lots of different techniques to get it right. It all came down to understanding what TensorFlow is doing to recognize patterns and features and give it what it needed to not key in on "unintentional" artifacts that crop up as a result of environmental situations (e.g. lighting).

Whenever you run into something like this, first search within the FTC Community forums - I bet someone else has run into your issue before. And, in fact, they have. The best discussion on this topic can be found here:
https://ftc-community.firstinspires.org/t/proper-process-for-object-detection/277

Once you've read the entire thread, and have followed and read links within my initial post, if you have additional questions please feel free to create a new topic on the FTC Community forums and ask.

-Danny
